File: src/main/java/com/toy/shopwebmvc/constant/Role.java
package com.toy.shopwebmvc.constant;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum Role {

    ROLE_ADMIN("어드민"),
    ROLE_BUYER("소비자"),
    ROLE_SELLER("판매자");

    private final String description;
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/constant/Role.kt
package spring.web.kotlin.constant

import spring.web.kotlin.exception.CommonException

enum class Role(
    val description: String
) {
    ROLE_ADMIN("어드민"),
    ROLE_BUYER("소비자"),
    ROLE_SELLER("판매자");

    companion object {
        fun of(name: Any?) = entries.find { it.name == name }
            ?: throw CommonException(ApiResponseCode.BAD_REQUEST)
    }
}